bin/i18n/sync-in.rb

```diff
@@ -109,6 +110,31 @@ def localize_docs
   end
 end
 
+# These files are synced in using the `bin/i18n-codeorg/in.sh` script.
+def localize_external_sources
+  puts "Preparing external sources"
+  external_sources_dir = File.join(I18N_SOURCE_DIR, "external-sources")
+
+  # ml-playground files
+  # These are overwritten in this format so the properties that use
+  # arrays have unique identifiers for translation.
+  dataset_files = File.join(external_sources_dir, 'ml-playground', 'datasets', '*')
+  Dir.glob(dataset_files).each do |dataset_file|
+    original_dataset = JSON.parse(File.read(dataset_file))
+
+    # Currently only including fields for translation.
+    # Use field id as unique identifier.
+    final_dataset = Hash.new
+    final_dataset["fields"] = Hash.new
+    dataset_fields = original_dataset["fields"]
+    dataset_fields.each {|field| final_dataset["fields"][field["id"]] = field}
```

Comment:
Maybe a curiosity, but this should be equivalent to the above lines, I think. I generally do not do a `Hash.new` instead of just a `final_dataset = {}`

```
final_dataset = {"fields" => original_dataset["fields"].map{|field| [field["id"], field]}.to_h}
```

Reply:
Ahh this is clean ✨ . I support it